crates/bpf-common/src/containers/mod.rs

```diff
@@ -0,0 +1,283 @@
+                };
+                let image = config.config.image;
+                let image_digest = config.image_digest;
+
+                Ok(Self {
+                    id,
+                    name,
+                    image,
+                    image_digest,
+                })
+            }
+            ContainerId::Libpod(container_id) => {
+                let (db_config, db_path, image_store_path) = find_libpod_container(&container_id)?;
+                let config: LibpodConfig =
+                    serde_json::from_str(&db_config.json).map_err(|source| {
+                        ContainerError::ParseConfigDB {
+                            source,
+                            path: db_path,
+                        }
+                    })?;
+                let image_id = config.rootfs_image_id;
+
+                let image_store_file =
+                    File::open(&image_store_path).map_err(|source| ContainerError::ReadFile {
+                        source,
+                        path: image_store_path.clone(),
+                    })?;
+                let reader = BufReader::new(image_store_file);
+
+                let images: Vec<LibpodImageConfig> =
+                    serde_json::from_reader(reader).map_err(|source| {
+                        ContainerError::ParseConfigFile {
+                            source,
+                            path: image_store_path.clone(),
+                        }
+                    })?;
+                let image = images.iter().find(|image| image.id == image_id).ok_or(
+                    ContainerError::ImageNotFound {
+                        id: image_id,
+                        path: image_store_path,
+                    },
+                )?;
+
+                Ok(Self {
+                    id: container_id,
+                    name: config.name,
+                    image: config.rootfs_image_name,
+                    image_digest: image.digest.clone(),
+                })
+            }
+        }
+    }
+}
+
+fn find_libpod_container(
+    container_id: &str,
+) -> Result<(LibpodDBContainerConfig, PathBuf, PathBuf), ContainerError> {
+    for (db_path, image_store_path) in find_existing_libpod_files()? {
+        if let Some(container) = find_libpod_container_in_db(&container_id, &db_path)? {
+            return Ok((container, db_path, image_store_path));
+        }
+    }
+
+    Err(ContainerError::ContainerNotFound {
+        id: container_id.to_owned(),
+    })
+}
+
+fn find_existing_libpod_files() -> Result<Vec<(PathBuf, PathBuf)>, ContainerError> {
+    let mut db_files = Vec::new();
+
+    let db_root_path = PathBuf::from(LIBPOD_DB_ROOT_PATH);
+    let image_store_root_path = PathBuf::from(LIBPOD_IMAGE_STORE_PATH);
+    if db_root_path.exists() && image_store_root_path.exists() {
+        db_files.push((db_root_path, image_store_root_path));
+    }
+
+    let file = File::open(ETC_PASSWD).map_err(|source| ContainerError::ReadFile {
```

Comment:
And same here - this is parsing /etc/passwd to get the potential /home directories where we look up for podman files.

The only alternative which comes up to my mind is keeping an internal mapping of users and homedirs. We could have a probe detecting new user creation. Not very easy, but also not very difficult to implement, but I didn't want to jump to it immediately before discussing.